### PR TITLE
Docs: Flux support in influxdb data source in 7.1

### PR DESCRIPTION
**Pull request by kylebrandt**
as per #25547 , documenting the new Flux query option in the InfluxDB datasource.

### CODE REVIEW

**Comment by bergquist** [comment]
Moving this to the 7.1 stable release

**Review by ryantxu — APPROVED** [review]
@kylebrandt -- I updated them...not sure the best process here

File: docs/sources/features/datasources/influxdb.md [hunk 1]
```diff
@@ -117,6 +121,61 @@ You can switch to raw query mode by clicking hamburger icon and then `Switch edi
 You can remove the group by time by clicking on the `time` part and then the `x` icon. You can
 change the option `Format As` to `Table` if you want to show raw data in the `Table` panel.
 
+## Flux support
+
+> Starting in v7.1 grafana will execute Flux queries in addition to InfluxQL queries
+
```
Comment (kylebrandt) [inline docs/sources/features/datasources/influxdb.md:127]: Should we add something that is beta? @ryantxu I believe you said there is a beta flag for this somewhere now in Grafana.

Comment (ryantxu) [reply docs/sources/features/datasources/influxdb.md:127]: There is a very clear flag in the UI... not sure it is necessary in the docs.  I'm not sure what we do in general here, and think either wasy is fine:

image

File: docs/sources/features/datasources/influxdb.md [hunk 2]
```diff
@@ -20,9 +20,13 @@ Grafana ships with a feature-rich data source plugin for InfluxDB. The plugin in
 2. In the side menu under the `Dashboards` link you should find a link named `Data Sources`.
 3. Click the `+ Add data source` button in the top header.
 4. Select *InfluxDB* from the *Type* dropdown.
+5. Select *InfluxQL* or *Flux* from the `Query Language` dropdown.
```
Comment (oddlittlebird) [inline docs/sources/features/datasources/influxdb.md:23]:
```suggestion
5. Select *InfluxQL* or *Flux* from the `Query Language` list.
```

File: docs/sources/features/datasources/influxdb.md [hunk 3]
```diff
@@ -20,9 +20,13 @@ Grafana ships with a feature-rich data source plugin for InfluxDB. The plugin in
...
 
 > NOTE: If you're not seeing the `Data Sources` link in your side menu it means that your current user does not have the `Admin` role for the current organization.
```
Comment (oddlittlebird) [inline docs/sources/features/datasources/influxdb.md:25]:
```suggestion
> **Note:** If you're not seeing the `Data Sources` link in your side menu it means that your current user does not have the `Admin` role for the current organization.
```

File: docs/sources/features/datasources/influxdb.md [hunk 4]
```diff
@@ -117,6 +121,61 @@ You can switch to raw query mode by clicking hamburger icon and then `Switch edi
 You can remove the group by time by clicking on the `time` part and then the `x` icon. You can
 change the option `Format As` to `Table` if you want to show raw data in the `Table` panel.
 
+## Flux support
+
+> Starting in v7.1 grafana will execute Flux queries in addition to InfluxQL queries
```
Comment (oddlittlebird) [inline docs/sources/features/datasources/influxdb.md:126]:
```suggestion
> Starting in v7.1, Grafana can execute Flux queries.
```

File: docs/sources/features/datasources/influxdb.md [hunk 5]
```diff
@@ -117,6 +121,61 @@ You can switch to raw query mode by clicking hamburger icon and then `Switch edi
...
+
+The client supports flux running on InfluxDB 1.8+.  See [1.8 compatibility](https://github.com/influxdata/influxdb-client-go/#influxdb-18-api-compatibility) for more details and connection details
```
Comment (oddlittlebird) [inline docs/sources/features/datasources/influxdb.md:128]:
```suggestion
The client supports Flux running on InfluxDB 1.8+.  See [1.8 compatibility](https://github.com/influxdata/influxdb-client-go/#influxdb-18-api-compatibility) for more information and connection details.
```

File: docs/sources/features/datasources/influxdb.md [hunk 6]
```diff
@@ -117,6 +121,61 @@ You can switch to raw query mode by clicking hamburger icon and then `Switch edi
...
+
+
+Name | Description
+------------ | -------------
+*Url* | The HTTP protocol, IP address and port of your InfluxDB API (InfluxDB 2.0 API port is by default 9999)
```
Comment (oddlittlebird) [inline docs/sources/features/datasources/influxdb.md:133]: Can you change this to "URL" in both the docs and the UI?

File: docs/sources/features/datasources/influxdb.md [hunk 7]
```diff
@@ -117,6 +121,61 @@ You can switch to raw query mode by clicking hamburger icon and then `Switch edi
...
+*Organization* | The [Influx organization](https://v2.docs.influxdata.com/v2.0/organizations/) that will be used for Flux queries.  This is also used to for the `v.organization` query macro
+*Token* | The authentication token used for Flux queries. With Influx 2.0, use the [influx authentication token to function](https://v2.docs.influxdata.com/v2.0/security/tokens/create-token/).  For influx 1.8, the token is `username:password` 
+*Default Bucket* | The [Influx bucket](https://v2.docs.influxdata.com/v2.0/organizations/buckets/) that will be used for the `v.defaultBucket` macro in Flux queries
+
+You can use the [Flux query and scripting language](https://www.influxdata.com/products/flux/). Grafana's Flux query editor is a text editor for raw Flux queries with Macro support.
+
+
+### Supported Macros
+
+The macros are designed to easily support copy+paste from [Chronograph](https://www.influxdata.com/time-series-platform/chronograf/)
```
Comment (oddlittlebird) [inline docs/sources/features/datasources/influxdb.md:143]:
```suggestion
The macros support copying and pasting from [Chronograph](https://www.influxdata.com/time-series-platform/chronograf/).
```

**Review by oddlittlebird — CHANGES_REQUESTED** [review]

File: docs/sources/features/datasources/influxdb.md [hunk 8]
```diff
@@ -29,9 +29,13 @@ Name | Description
 *Default* | Default data source means that it will be pre-selected for new panels.
 *Url* | The HTTP protocol, IP address and port of your InfluxDB API (InfluxDB API port is by default 8086)
 *Access* | Server (default) = URL needs to be accessible from the Grafana backend/server, Browser = URL needs to be accessible from the browser.
+*Enable flux* | Enables support for the [Flux query and scripting language](https://www.influxdata.com/products/flux/). Requires that the Token field be set a valid [Influx authentication token to function](https://v2.docs.influxdata.com/v2.0/security/tokens/create-token/)
```
[the remark this answers is not in the document]
Comment (ryantxu) [reply docs/sources/features/datasources/influxdb.md]: @oddlittlebird -- I'm not sure what you are looking at... ~1/2 of your comments are against outdated content.  I applied the ones that can merge though :man_shrugging:

[the remark this answers is not in the document]
Comment (oddlittlebird) [reply docs/sources/features/datasources/influxdb.md]: YEah, had some weirdness on my end. I commented and THOUGHT I had applied the review last week. Apparently not...but the comments were still waiting.

**Review by oddlittlebird — APPROVED** [review]

File: docs/sources/features/datasources/influxdb.md [hunk 9]
```diff
@@ -117,6 +121,48 @@ You can switch to raw query mode by clicking hamburger icon and then `Switch edi
...
 
+## Flux support
+
+> Only available in Grafana v7.1+.
+
+If you have enabled Flux and set up a token for auth in the data source configuration, then you can use the [Flux query and scripting language](https://www.influxdata.com/products/flux/). Grafana's Flux query editor is a text editor for raw Flux queries with Macro support.
+
+### Supported Macros
+
+Macro example | Description
```
[the remark this answers is not in the document]
Comment (ryantxu) [reply docs/sources/features/datasources/influxdb.md:145]: :+1: -- already capitalized in the UI (unless you see it lowercase somewhere)

**Pull request merged**